crates/autopilot/src/run_loop.rs

```diff
@@ -52,13 +50,20 @@ pub struct RunLoop {
     pub solve_deadline: Duration,
     pub in_flight_orders: Arc<Mutex<Option<InFlightOrders>>>,
     pub liveness: Arc<Liveness>,
+    pub synchronization: RunLoopMode,
 }
 
 impl RunLoop {
     pub async fn run_forever(self) -> ! {
         let mut last_auction = None;
         let mut last_block = None;
         loop {
+            if let RunLoopMode::SyncToBlockchain = self.synchronization {
+                let _ = ethrpc::block_stream::next_block(self.eth.current_block()).await;
+            } else {
+                tokio::time::sleep(Duration::from_secs(1)).await;
+            }
+
```

Comment:
Shouldn't we check if `self.eth.current_block().borrow()` is still `current_block`? Only if this is the case, we should await the next block, but otherwise immediately start the next runloop. 

This way we aren't slower than the status quo (we might start half-way into the block if the `driver.settle` request finished mid block, but I assume this will get aligned later on).

Reply:
Right now we are in most cases idle for 12 seconds, because the previous runloop ends when block x is seen (this triggers the solution to be settled) and then we only start the next auction in x+1 which seems wasteful.

Reply:
Yeah this is what I alluded to (but not explained thoroughly) in the description:
> At the moment this will actually slow us down because the solution submission is not yet prepared for 1 auction per block [...]

Also already brought up [here](https://github.com/cowprotocol/services/pull/2884#discussion_r1715139834).
Overall I don't see the value in "synchronizing" to the block stream and then ignoring the synchronization mechanism 100% of the time (at least in the status quo). And for the future we should focus our efforts on ensuring that the runloop (i.e. submission) ends at a reasonable time instead of working around timing issues by ignoring the synchronization mechanism which I think you referred to with:
> but I assume this will get aligned later on

Reply:
> And for the future we should focus our efforts on ensuring that the runloop (i.e. submission) ends at a reasonable time

Imo a reasonable time is when it saw the next block. It can then check if the settlement transaction was in it and end the current run loop at which point a new run loop should start right away. Unless you want to move the `settle` call into a background task that gets "finished" outside the core runloop.

Reply:
So as part of updating solvable orders it also runs on settlement updater for the most recent block?